drivers/sensor/lis2dh/lis2dh_trigger.c

```diff
@@ -121,7 +121,7 @@ static inline void setup_int2(const struct device *dev,
 
 	gpio_pin_interrupt_configure_dt(&cfg->gpio_int,
 					enable
-					? GPIO_INT_LEVEL_ACTIVE
+					? GPIO_INT_EDGE_RISING
```

Comment:
Please see #39880

cc: @tautologyclub

Reply:
I would not be so sure this is the way forward for hp filter. Speaking from direct experience, the HP filter is something you want to have runtime control over.

I will read through the PR in full and elaborate, gimme a few minutes

Reply:
Ok so the interrupt reconfiguration clearly has merit, but 1. clearly does not belong in this PR and 2. is clearly incomplete. Start a new PR reverting #39880 or making it configurable or whatever. Let's focus on HP filtering in this PR.

Reply:
Agree that 2 changes should be separated (In my project both was required to get normal behavior). I'm using LIS3DH, but as I can see they share the same register set.  Will try with reverting https://github.com/zephyrproject-rtos/zephyr/pull/39880 and see how it goes. 
For the filter part, also agree that run-time configuration would be a preferred way, most likely by sensor attributes, so let me try it next few days.